team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
> Specifically: Do you have an example of something that works wrong if you leave out template.id === dexEntry.prevo || dexEntry.evos.includes(template.id)?

Yeah, sorry I missed that the first time. If you have Scyther on your team with Defog (which should be an unavailable egg move) and three other non-egg moves you get a valid team for [Gen 7] OU when that should not be the case. The pokemon Scyther is inheriting Defog from turns out to be Scizor, and that should not be possible.

> Do not execute checkLearnset for the breeder. I'd say just leave it like this and hardcode incompatibilities as they come up.

Alright, fair enough.